tests/test_stemmus_scope.py

```diff
@@ -0,0 +1,132 @@
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+import os
+import subprocess
+from PyStemmusScope import StemmusScope
+from PyStemmusScope import config_io
+from . import data_folder
+
+
+class TestWithDefaults:
+    @pytest.fixture
+    def model(self, tmp_path):
+        config_file = str(data_folder / "config_file_test.txt")
+        exe_file = Path(tmp_path) / "STEMUUS_SCOPE"
+
+        # create dummy exe file
+        with open(exe_file, "x", encoding="utf8") as dummy_file:
+            dummy_file.close()
+
+        yield StemmusScope(config_file, exe_file)
+
+    @pytest.fixture
+    def model_with_setup(self, model):
+        with patch("time.strftime") as mocked_time:
+            mocked_time.return_value = "2022-07-11-1200"
+
+            cfg_file = model.setup()
+            return model, cfg_file
+
+    def test_setup(self, model_with_setup):
+        model, cfg_file = model_with_setup
+
+        actual_input_dir = data_folder / "directories" / "input" / "XX-dummy_2022-07-11-1200"
+        actual_output_dir = data_folder / "directories" / "output" / "XX-dummy_2022-07-11-1200"
+        actual_cfg_file = str(actual_input_dir / "XX-dummy_2022-07-11-1200_config.txt")
+
+        assert actual_input_dir == Path(model.configs["InputPath"])
+        assert actual_output_dir == Path(model.configs["OutputPath"])
+        assert actual_cfg_file == cfg_file
+
+        # matlab log dir
+        assert os.environ['MATLAB_LOG_DIR'] == str(model.configs["InputPath"])
+
+    @patch("subprocess.run")
+    def test_run(self, mocked_run, model_with_setup):
+
+        actual_cfg_file = data_folder / "directories" / "input" / "XX-dummy_2022-07-11-1200_config.txt"
+        output = (
+            f"b'Reading config from {actual_cfg_file}\n\n "
+            "The calculations start now \r\n The calculations end now \r'"
+            )
+
+        mocked_run.return_value.stdout = output
+
+        model, cfg_file = model_with_setup
+        result = model.run()
+
+        expected = [f"{model.exe_file} {cfg_file}"]
+        mocked_run.assert_called_with(
+        expected, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, check=True,
+        )
+
+        # output of subprocess
+        assert result == output
```

Comment:
Currently the STEMMUS_SCOPE executable is not shipped with PyStemmusScope, so the actual model run can't be tested. Will this change in the future?

Reply:
The idea is that PyStemmusScope should work with different versions of the executable file. Also, because the source code is maintained in a different repository, the exe won't be shipped with PyStemmuScope. For test purposes, it is possible to create some test datasets later.